lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -99,8 +99,24 @@ func (i *InteractionHandlersImpl) ContainerResizeHandler(params interaction.Cont
 }
 
 func (i *InteractionHandlersImpl) ContainerSetStdinHandler(params interaction.ContainerSetStdinParams) middleware.Responder {
-	log.Printf("Attempting to get ssh session for container %s stdin", params.ID)
-	sshConn, err := i.attachServer.Get(context.Background(), params.ID, interactionTimeout)
+	var ctxDeadline time.Time
+	var timeout time.Duration
+
+	// Calculate the timeout for the attach if the caller specified a deadline
+	if params.Deadline != nil {
+		ctxDeadline = time.Time(*params.Deadline)
+		timeout = ctxDeadline.Sub(time.Now())
+		log.Printf("Attempting to get ssh session for container %s stdin with deadline %s", params.ID, ctxDeadline.Format(time.UnixDate))
+		if timeout < 0 {
+			e := &models.Error{Message: fmt.Sprintf("Deadline for stdin already passed for container %s", params.ID)}
```

Comment:
The idea was that we wanted to start a "timer" from when a caller performs an attach on the personality server side.  That is total time we're going to allow, inclusive of the portlayer.  That includes the time it takes for the ssh server to finish.  Since this is total overall time, defined by the personality server side, we need the portlayer to also abide by that time so giving it a hard deadline is the only way to remove any other overhead.